File: chroma-league/src/main/java/com/bonepl/chromaleague/hud/parts/resource/VladimirBloodPoolBar.java
package com.bonepl.chromaleague.hud.parts.resource;

import com.bonepl.chromaleague.hud.parts.ProgressBar;
import com.bonepl.chromaleague.state.GameStateHelper;
import com.bonepl.razersdk.animation.AnimatedFrame;
import com.bonepl.razersdk.color.Color;
import com.bonepl.razersdk.animation.Frame;
import com.bonepl.razersdk.color.StaticColor;

public class VladimirBloodPoolBar extends AnimatedFrame {
    private int previousResourcePercentage;

    @Override
    public Frame getFrame() {
        final int currentResourcePercentage = GameStateHelper.getResourcePercentage();
        if (currentResourcePercentage > previousResourcePercentage) {
            if (currentResourcePercentage < 50) {
                addAnimationFrame(new ProgressBar(ResourceBars.getResourceBarKeys(), currentResourcePercentage, StaticColor.WHITE));
            } else {
                addAnimationFrame(new ProgressBar(ResourceBars.getResourceBarKeys(), currentResourcePercentage, StaticColor.YELLOW));
            }
        } else {
            addAnimationFrame(new ProgressBar(ResourceBars.getResourceBarKeys(), currentResourcePercentage, StaticColor.RED));
        }
        previousResourcePercentage = currentResourcePercentage;
        return super.getFrame();
    }
}

File: src/hud/parts/resource/VladimirBloodPoolBar.ts
import {AnimatedFrame} from '../../../razer-sdk/animation/AnimatedFrame.js';
import {Frame} from '../../../razer-sdk/animation/Frame.js';
import {StaticColor} from '../../../razer-sdk/color/StaticColor.js';
import {ProgressBar} from '../ProgressBar.js';
import {getResourceBarKeys} from './ResourceBars.js';
import {getResourcePercentage} from '../../../state/GameStateHelper.js';

export class VladimirBloodPoolBar extends AnimatedFrame {
    private previousResourcePercentage = 0;

    pollFrame(): Frame {
        const currentResourcePercentage = getResourcePercentage();
        if (currentResourcePercentage > this.previousResourcePercentage) {
            if (currentResourcePercentage < 50) {
                this.addAnimationFrame(new ProgressBar(getResourceBarKeys(), currentResourcePercentage, StaticColor.WHITE));
            } else {
                this.addAnimationFrame(new ProgressBar(getResourceBarKeys(), currentResourcePercentage, StaticColor.YELLOW));
            }
        } else {
            this.addAnimationFrame(new ProgressBar(getResourceBarKeys(), currentResourcePercentage, StaticColor.RED));
        }
        this.previousResourcePercentage = currentResourcePercentage;
        return super.pollFrame();
    }
}
